### PR TITLE
random: General fixes

### LINKED ISSUE #62720: Thread safety of the random subsystem
**Describe the bug**

All of the threads in Zephyr RTOS share a common random number subsystem instance to generate random numbers and CSPRNGs. RNG and CSPRNG backends are selected independently.

Some of these backends are thread-safe:

* ctr_drbg (irq_lock)
* timer (spin_lock)

Some are not:

* xoshiro

For others, it depends on the device driver thread safety:

* entropy_device

**Expected behavior**

At least all CSPRNG backends are thread-safe.

It seems currently there are two CSPRNG backends available:

* ctr_drbg
* entropy_device

*ctr_drbg* is already thread-safe. For *entropy_device* we would need to either add some synchronization primitives to the *entropy_device* backend implementation or require all entropy device driver implementations to be thread-safe. I checked a few selected driver implementations and they seem already thread-safe by using spin_locks or irq_locks. So maybe it's just a matter of documenting this requirement in entropy device driver API docs to fix the problem I'm describing.

**Impact**
Security

### PR DESCRIPTION
**Pull request by ceolin**
- Remove unused variable
- Make csprng thread safe
- Document driver interface requirement

### CODE REVIEW

File: include/zephyr/drivers/entropy.h [hunk 1]
```diff
@@ -33,6 +33,9 @@ extern "C" {
  * @typedef entropy_get_entropy_t
  * @brief Callback API to get entropy.
  *
+ * @note This API be thread safe since it can be used
```
Comment (nashif) [inline include/zephyr/drivers/entropy.h:36]: This API is...?

Comment (d3zd3z) [reply include/zephyr/drivers/entropy.h:36]: Probably want to just say something like "This call is thread safe". Perhaps "This call is thread safe, to satisfy requirements of the random subsystem."

**Comment by d3zd3z** [comment]
Other than the comment fix, this looks good.

File: include/zephyr/drivers/entropy.h (hunk 1 above)
Comment (ceolin) [reply include/zephyr/drivers/entropy.h:36]: argg... I missed the "This API *needs to*"

**Comment by ceolin** [comment]
@d3zd3z @nashif can you take a look again please

**Review by nashif — APPROVED** [review]

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**